synapseclient/client.py

```diff
@@ -1134,13 +1135,199 @@ def _getEntityBundle(self, entity, version=None, bitFlags=0x800 | 0x400 | 0x2 |
+
+        results_found = False
+        for result in results:
+            results_found = True
+
+            fmt_fields = {'name' : result['entity.name'],
+                          'id' : result['entity.id'],
+                          'padding' : ' ' * indent,
+                          'slash_or_not' : '/' if is_container(result) else ''}
+            fmt_string = "{id}"
+
+            if long_format:
+                fmt_fields['createdOn'] = utils.from_unix_epoch_time(result['entity.createdOn']).strftime("%Y-%m-%d %H:%M")
+                fmt_fields['createdBy'] = self._get_user_name(result['entity.createdByPrincipalId'])[:18]
+                fmt_fields['version']   = result['entity.versionNumber']
+                fmt_string += " {version:3}  {createdBy:>18} {createdOn}"
+            if show_modified:
+                fmt_fields['modifiedOn'] = utils.from_unix_epoch_time(result['entity.modifiedOn']).strftime("%Y-%m-%d %H:%M")
+                fmt_fields['modifiedBy'] = self._get_user_name(result['entity.modifiedByPrincipalId'])[:18]
+                fmt_string += "  {modifiedBy:>18} {modifiedOn}"
+
+            fmt_string += "  {padding}{name}{slash_or_not}\n"
+            out.write(fmt_string.format(**fmt_fields))
+
+            if (indent==0 or recursive) and is_container(result):
+                self._list(result['entity.id'], recursive=recursive, long_format=long_format, show_modified=show_modified, indent=indent+2, out=out)
+
+        if indent==0 and not results_found:
+            out.write('No results visible to {username} found for id {id}\n'.format(username=self.username, id=id_of(parent)))
+
+
+    ############################################################
+    ##                 Copy Functions                         ##
+    ############################################################
+
+    def copy(self, entity, destinationId=None, copyWiki=True, onlyCopyWiki=False, **kwargs):
+        """
+        Copies synapse entities including the wikis
+
+        :param entity:          A synapse entity ID
+
+        :param destinationId:   Synapse ID of a folder/project that the copied entity is being copied to
+
+        :param copyWiki:        Determines whether the wiki of the entity is copied over
+                                Default is True
+
+        :param onlyCopyWiki:    Determines whether only the wiki of the entity is copied
+                                Default is False
+
+        :param copyTable:       Determines whether tables are copied
+                                Default is True
+
+        :param kwargs:          Parameters that can be passed to the hidden copy functions called
+        """
+        mapping = kwargs.get('mapping', dict())
+        updateLinks = kwargs.get('updateLinks', True)
+        updateSynIds = kwargs.get('updateSynIds', True)
+        if onlyCopyWiki:
+            self._copyWiki(entity, destinationId, updateLinks=updateLinks, updateSynIds=updateSynIds, entityMap=mapping)
```

Comment:
expose syn.copyWiki(...) for when we only want to copy wiki. Then syn.copy(...) can optionally call syn.copyWiki(...).